builder/builder_test.go

```diff
@@ -36,6 +51,79 @@ func (*queryAll) String() string {
 	return "all"
 }
 
+func setupTestEnvironment(t *testing.T) testEnvironment {
```

Comment:
Can you move the new test helpers below the actual tests (`TestBuild`, `TestRollback`, etc) for readability?